src/RandomMapsChooser.as

```diff
@@ -283,6 +289,7 @@ namespace RandomMapsChooser {
     }
 
     void GetMapsTillDone() {
+        dictionary blacklistDict = ParseBlacklistToDict(blacklist);
```

Comment:
use handles in situations like this to avoid duplicating the underlying data structure.